voice/eo/eo_tts.js

```diff
@@ -0,0 +1,660 @@
+	}
+}
+
+function follow_street(streetName) {
+	if ((streetName["toDest"] == "" && streetName["toStreetName"] == "" && streetName["toRef"] == "") || Object.keys(streetName).length == 0 || !tts) {
+		return "";
+	} else if (streetName["toStreetName"] == "" && streetName["toRef"] == "") {
+		return dictionary["to"] + " " + streetName["toDest"];
+	} else if (streetName["toRef"] == streetName["fromRef"] && streetName["toStreetName"] == streetName["fromStreetName"] || 
+			(streetName["toRef"] == streetName["fromRef"] && streetName["toStreetName"] == "")) {
+		return dictionary["on"] + " " + assemble_street_name(streetName);
+	} else if (!(streetName["toRef"] == streetName["fromRef"] && streetName["toStreetName"] == streetName["fromStreetName"])) {
+		return dictionary["to"] + " " + assemble_street_name(streetName);
+	}
+}
+
+function turn(turnType, dist, streetName) {
+	if (dist == -1) {
+		return getTurnType(turnType) + " " + turn_street(streetName);
+	} else {
+		return dictionary["after"] + " " + distance(dist) + " " + getTurnType(turnType) + " " + turn_street(streetName); 
+	}
+}
+
+function take_exit(turnType, dist, exitString, exitInt, streetName) {
+	if (dist == -1) {
+		return getTurnType(turnType) + " " + dictionary["onto_exit"] + " " + getExitNumber(exitString, exitInt) + " " + take_exit_name(streetName);
+	} else {
+		return dictionary["after"] + " " + distance(dist) + " "
+			+ getTurnType(turnType) + " " + dictionary["onto_exit"] + " " + getExitNumber(exitString, exitInt) + " " + take_exit_name(streetName);
+	}
+}
+
+function take_exit_name(streetName) {
+	if (Object.keys(streetName).length == 0 || (streetName["toDest"] == "" && streetName["toStreetName"] == "") || !tts) {
+		return "";
+	} else if (streetName["toDest"] != "") {
+		return (tts ? ", " : " ") + streetName["toStreetName"] + " " + dictionary["toward"] + " " + streetName["toDest"];
+	} else if (streetName["toStreetName"] != "") {
+		return (tts ? ", " : " ") + streetName["toStreetName"];
+	} else {
+		return "";
+	}
+}
+
+function getExitNumber(exitString, exitInt) {
+	if (!tts && exitInt > 0 && exitInt < 18) {
+		return dictionary["the"] + nth_acc(exitInt) + " " + dictionary["exit_acc"];
+	} else if (tts) {
+		return dictionary["exit_acc"] + " " + exitString;
+	} else {
+		return dictionary["exit_acc"];
+	}
+}
+
+function getTurnType(turnType) {
+	switch (turnType) {
+		case "left":
+			return dictionary["left"];
+			break;
+		case "left_sh":
+			return dictionary["left_sh"];
+			break;
+		case "left_sl":
+			return dictionary["left_sl"];
+			break;
+		case "right":
+			return dictionary["right"];
+			break;
+		case "right_sh":
+			return dictionary["right_sh"];
+			break;
+		case "right_sl":
+			return dictionary["right_sl"];
+			break;
+		case "left_keep":
+			return dictionary["left_keep"];
+			break;
+		case "right_keep":
+			return dictionary["right_keep"];
+			break;
+	}
+}
+
+function then() {
+	return dictionary["then"] + " ";
```

Comment:
Here you may want to test if better intonation requires a preceding comma, i.e. `return (tts ? ", " : " ") + dictionary["then"] + " ";` Most languages have this to better pause these concetated prompts.

Reply:
Yeah, that makes sense.